Refactor clinicians tests for jsonapi

nmajor25 opened the pull request.

nmajor25 committed:
Refactor clinicians tests for jsonapi

nmajor25 commented on test/integration/clinicians/clinician-sidebar.js:
```diff
@@ -449,9 +459,9 @@ context('clinician sidebar', function() {
   specify('view clinician', function() {
```
Don't think this test does anything unique? It just checks the clinician name and disabled email in the clinician sidebar.

The same things are done in the `edit clinician` test. Unless I'm missing something...

paulfalgout replied:
yeah if we can remove this without coverage drop, we should.  edit was probably added later, duplicating this when it should have just been modified

nmajor25 replied:
This was removed and test coverage remained at 100% 👍

nmajor25 commented on test/support/api/workspaces.js:
```diff
@@ -23,6 +23,12 @@ export function getWorkspaces(data, { depth = 0 } = {}) {
+
+// Exporting only workspaces needed for testing variance
+export const workspaceOne = _.find(workspaces, { id: '11111' });
+export const workspaceTwo = _.find(workspaces, { id: '22222' });
```
This is for the `find in list` test above, where we need each clinician needs to only be part of one workspace.

Attempting to follow the same pattern in roles and teams support files.

nmajor25 commented on test/support/api/workspaces.js:
```diff
@@ -23,6 +23,12 @@ export function getWorkspaces(data, { depth = 0 } = {}) {
   return mergeJsonApi(resource, data, { VALID: { relationships: _.keys(defaultRelationships) } });
 }
 
+const workspaces = getResource(fxTestWorkspaces, TYPE);
```
I tried to do `const workspaces = getWorkspaces()` here but the `clinicians: getRelationship(getClinicians({}, { depth })),` was throwing an error in the `getWorkspaces()` function.

Not really sure why.

paulfalgout replied:
This is a dependency issue..  I don't know how to fix it because we `import` all of the files in one file..  I think we'd have to always export functions..

nmajor25 replied:
So leave this how I have it for now?

paulfalgout commented on test/integration/clinicians/clinician-sidebar.js:
```diff
@@ -1,10 +1,12 @@
+import { getClinician, getCurrentClinician } from 'support/api/clinicians';
 
 const testClinician = {
   id: '1',
```
I think rather than storing a pojo here we could get `getClinician` at this point?  Then we don't need to set the `id` and we can test with `testClinician.id` in the expects

nmajor25 replied:
I just updated this to be `const testClinician = getClinician({})` and then used that throughout the tests.

paulfalgout commented on test/integration/clinicians/clinicians-all.js:
```diff
@@ -1,24 +1,38 @@
-        fx.data[1].attributes.last_active_at = null;
+        fx.data = [
+          getClinician({
+            id: '1',
```
I don't think these ids are needed

nmajor25 replied:
If I don't set the `id` here, an error is thrown:

![Screenshot 2024-01-09 at 10 37 10 AM](https://github.com/RoundingWell/care-ops-frontend/assets/35355575/b9b5f07d-9367-4d0f-99b2-f84d5e78ca95)

paulfalgout replied:
oh ha.  well that's why I put that error in there because this one is unique

paulfalgout commented on test/integration/clinicians/clinicians-all.js:
```diff
@@ -186,18 +200,30 @@ context('clinicians list', function() {
+              enabled: true,
+            },
+            relationships: {
+              workspaces: [getRelationship([workspaceTwo])],
```
there's an extra set of arrays here..

nmajor25 replied:
Oops I was playing around with something previously and forgot to remove those. Should be removed now.

nmajor25 committed:
Remove 'clinician-sidebar' => 'view clinician' test

This test contains things already tested in the 'edit clinician' test in this same file
